hist/histpainter/src/THistPainter.cxx

```diff
@@ -3416,6 +3416,20 @@ void THistPainter::ExecuteEvent(Int_t event, Int_t px, Int_t py)
    TAxis *yaxis    = fH->GetYaxis();
    Int_t dimension = fH->GetDimension();
 
+   // In case of option SAME the axis must be the ones of the first drawn histogram
+   TString IsSame = fH->GetDrawOption();
+   IsSame.ToLower();
+   if (IsSame.Index("same")>=0) {
+      TH1 *h1;
+      TIter next(gPad->GetListOfPrimitives());
+      while ((h1 = (TH1 *)next())) {
+         if (!h1->InheritsFrom(TH1::Class())) continue;
```

Comment:
But what if this histogram is also drawn with option "SAME"? Don't you need to keep going until you find one that doesn't have "SAME" set?

Reply:
Seems to me it is correct. The first in the list of primitives defines the X and Y ranges of the plot. It cannot have the option SAME. The histograms are displayed in order. So if the 1st in the list has option SAME it means it will not define the plot ranges .. it will use undefined ranges and the plot will be very likely wrong. If you have a macro illustrating this case I'll be happy to try it.